website/blog/2022-01-21-components-and-blocks.md

```diff
@@ -0,0 +1,54 @@
+- displaying a photo with the with the photographer's name
+- and many more...
+
+As you can already see, there are many ways you can use this component. What helps even more to expand the usability of this component are `attributes`. Lets say we have the following attributes:
+- `imagePosition` - left, right, top, bottom
+- `imageShape` - rectangular, circular
+- `imageWidth` - 25%, 50%, 75%
+- `headingColor` - black, blue, red
+- `headingSize` - small, medium, large
+
+These attributes give you the option to modify the look of the component, so this specific component can be used for the following:
+
+- **related posts** - rectangular image on the left side, 25% image width, heading takes up the remaining 75%, medium heading colored in blue
+- **user avatar** - circular image on the left side, 25% image width, small heading coloured in black
+- **photo of the day** - rectangular image on the right side which takes 75% width and photographer's name is written in medium blue text
+- **featured posts** - rectangular image on top, large heading in red color
+
+You even have the option to add conditional logic, e.g. to have the option for `imageWidth` only when `imagePosition` is set to left or right. And when `imagePosition` is set to top or bottom, it defaults to 100% width.
+
+### Block
+
+So, what about blocks? Blocks can consist of one or more components and, most importantly, allow you to use them in editor.
+**Blocks are what gives components context.**
+
+Now that we have our component, let's call it `card-simple`, we can use it in a `Featured Posts` block. We already have predefined attributes how our component should look when used in this block defined in manifest.json, but we also need another attribute which will allow us to select which posts will be displayed.
+
+In order to do that, we add a new attribute called `featuredPosts` which defines which posts will be displayed in our block. For each selected post, another `card-simple` component is added and featured image and post title are dynamically added to its image and heading components. More about the query logic and adding new options in a block will be covered in a future chapter.
+
+You may be wondering, can you have a block which allows you to use all options of the `card-simple` component? Of course! You can create a block which allows you to set all available options in editor and manually add content to image and heading. There are also block variations, which allow you to set up predefined attributes for a block. More about setting up block variations may be covered in one of the future chapters.
+
+### Conclusion
+
+Hopefully these examples helped you get a better understanding of the difference between components and blocks. It may take a bit more time or practical use to truly understand the difference and the reason behind it. Our next chapter will walk you through downloading one of our components and blocks with WP CLI and modifying it.
```

Comment:
```suggestion
Hopefully, these examples helped you get a better understanding of the difference between components and blocks. It may take a bit more time or practical use to understand the difference and reasoning behind it. Our next chapter will walk you through downloading one of our components and blocks with WP CLI and modifying it.
```